ezidapp/management/commands/proc-search-indexer.py

```diff
@@ -35,6 +43,75 @@ class Command(ezidapp.management.commands.proc_base.AsyncProcessingCommand):
     setting = 'DAEMONS_SEARCH_INDEXER_ENABLED'
     queue = ezidapp.models.async_queue.SearchIndexerQueue
 
+    # note, the method is an overridden version to allow for occasional retries of supposedly "permanent" errors
+    # such as network errors or a service being temporarily down or unresponsive.
+
+    # It is kind of kludgy since it uses an infrequently running check every 5 minutes (300 seconds)
+    # based on (current_time_in_seconds - submit_time_in_seconds) % 300 == 0 for up to 1 day since initial queuing.
+
+    # It's possible it may not hit this exact remainder exactly every 5 minutes since
+    # something may be running and not wake up at exactly 5 minute marks, but it will retry sometime soon since the
+    # as though the default wake-up frequency for checking is every second
+
+    # it may recheck up to 288 times (1 day / 5 minutes) before giving up on a task and leaving it as
+    # a permanent error.
+
+    # For the future we may want to consider a more sophisticated retry mechanism that maintains some
+    # retry logic and state in the database for these queue models as well as more sophisticated notifications
+    # on failures or a reporting system
+
+    # this retry could break if the sleep time is changed to be longer than 1 second and might not trigger ever if the
+    # wake up doesn't happen during the second it would trigger
+    def run(self):
+        """Run async processing loop forever.
+
+        The async processes that don't use a queue based on AsyncQueueBase must override
+        this to supply their own loop.
+
+        This method is not called for disabled async processes.
+        """
+        assert self.queue is not None, "Must specify queue or override run()"
+
+        while not self.terminated():
+            self._possibly_reset_errors_for_retry()
+            qs = self.queue.objects.filter(status=self.queue.UNSUBMITTED, ).order_by(
```

Comment:
How about add a timer to control when to include the failed transactions in the query. This way we do not need to update the status from 'F' to 'U'.

Reply:
I can do this if you want by running a different query at different times.  I don't want to check every failed thing every second which I think is wasteful and resource intensive.

However, in the long term I think the queue has some general problems for the future and it would be nice to solve these retry and also notification problems in a general way for all queues.  We also probably want to add additional dates to all database tables and some other things around notifications to make a better solution.  I'm not trying to fix everything at once and I also wonder if EZID would be better off using an off-the-shelf queuing/worker software (like Celery) rather than making it's own solution from scratch, but maybe there are reasons.

I don't particularly see resetting the state as a problem since if it fails again it will immediately revert to the failed state when it's tried milliseconds or 1 second later.

We also don't have a history of every retry in the queue, which IDK, maybe you want, but that is a larger redesign, I'd think.

I guess I'm trying to understand the reasoning behind the changes you're asking for since the end state will be the same.  In a few miliseconds it will retry again and succeed or fail and the information in the table will be updated anyway.